assets/js/googlesitekit/datastore/user/key-metrics.js

```diff
@@ -351,16 +353,67 @@ const baseSelectors = {
 	} ),
 
 	/**
-	 * Gets key metrics settings.
+	 * Gets key metrics settings, taking into account whether the user has view-only access.
+	 * If the user has view-only access and the custom dimensions required by the selected widgets are unavailable,
+	 * the widget slugs will be removed from the settings. If only one widget remains after filtering, the widget slugs
+	 * will be an empty array to prevent hiding the widget area.
 	 *
 	 * @since 1.103.0
+	 * @since n.e.x.t Checks for view-only access and adjusts the `widgetSlugs` accordingly.
 	 *
 	 * @param {Object} state Data store's state.
 	 * @return {(Object|undefined)} Key metrics settings. Returns `undefined` if not loaded.
 	 */
-	getKeyMetricsSettings( state ) {
-		return state.keyMetricsSettings;
-	},
+	getKeyMetricsSettings: createRegistrySelector( ( select ) => ( state ) => {
+		const keyMetricsSettings = state.keyMetricsSettings;
+
+		const isAuthenticated = select( CORE_USER ).isAuthenticated();
+		const canManageOptions = select( CORE_USER ).hasCapability(
+			PERMISSION_MANAGE_OPTIONS
+		);
+
+		const isViewOnly = ! isAuthenticated && ! canManageOptions;
```

Comment:
Hey @hussain-t, are you sure we need to check for the `manage_options` permission here?

Looking at the backend code where we set the `data-view-only` attribute that underpins the `useViewOnly()` hook and so forth, it's only reliant on the authenticated state (well, and the `VIEW_DASHBOARD` permission, but we can assume we're on the Dashboard already here):

https://github.com/google/site-kit-wp/blob/1ee627f779b1ce0ea24580b8bf8f8ecee8670416/includes/Core/Admin/Screens.php#L371-L379

Also, it should be implicit that an authenticated user has the `manage_options` permission as that's what the `AUTHENTICATE` permission maps to:

https://github.com/google/site-kit-wp/blob/1ee627f779b1ce0ea24580b8bf8f8ecee8670416/includes/Core/Permissions/Permissions.php#L150-L152

Maybe I am missing something, but I think we should be able to simplify things a bit by just checking for `isAuthenticated` here...

Reply:
That's a good point. Thanks, @techanvil 👍

Reply:
Updated ✅